test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -349,9 +389,23 @@ static void test_out_in_common(ipsec_test_flags *flags,
 			       const odp_crypto_key_t *auth_key_extra)
 {
 	odp_ipsec_sa_param_t param;
+	odp_ipsec_stats_t stats;
 	odp_ipsec_sa_t sa_out;
 	odp_ipsec_sa_t sa_in;
 
+	CU_ASSERT_NOT_EQUAL_FATAL(flags, NULL);
+
+	/*
+	 * ICV won't be generated for NULL AUTH. For AES-GCM & CHACHA-POLY
+	 * any data corruption would lead to alg error.
```

Comment:
I wonder if this is ok and according to the API spec or a bug in the implementation.

Reply:
I'm not sure if it's a bug or if that's the intended usage.

In linux-gen, when crypto status is not ok, 
```
if (cipher_status.alg_err)
    status->error.alg = 1;

if (auth_status.alg_err)
    status->error.auth = 1;
```

It's more like, when crypto & auth are separate operations, crypto(cipher_status.alg_err) will not be flagged for any data corruption. It's the auth check which allows us to detect errors. But when we do combined operation, crypto(cipher_status.alg_err) gets reported.

IMO, the above code snippet is not correct. It should be auth_err for either case. I can submit a fix for this and correct UT as well if you can confirm that's the intended usage.

Reply:
Yes, I think we should get auth error if the ICV is wrong. If you could submit a fix, that would be great.

Reply:
Done.

https://github.com/OpenDataPlane/odp/pull/1122